crates/net/network/src/transactions.rs

```diff
@@ -489,6 +545,32 @@ where
         }
     }
 
+    /// Return a list of all peer ids in the Peers hashmap
+    fn get_active_peers(&self) -> Vec<&PeerId> {
+        self.peers.keys().collect()
+    }
+
+    /// Return all the hashes known by given `peer_ids`
+    fn get_transaction_hashes(&self, peer_ids: Vec<PeerId>) -> Vec<&TxHash> {
+        let mut tx_set = HashSet::<&TxHash>::new();
+        peer_ids.into_iter().for_each(|id| {
+            self.get_peer_transaction_hashes(id).into_iter().for_each(|tx| {
+                tx_set.insert(tx);
+            })
+        });
+
+        tx_set.into_iter().collect()
+    }
+
+    /// Return all the transaction hashes know by the given peer
+    fn get_peer_transaction_hashes(&self, peer_id: PeerId) -> Vec<&TxHash> {
+        if let Some(peer) = self.peers.get(&peer_id) {
+            peer.transactions.iter().collect()
+        } else {
+            vec![]
+        }
+    }
```

Comment:
we don't really need these because this can be simplified

Reply:
Are you suggesting these should simply be inside `on_command`? Can you confirm these have the desired functionality before I move them?

Reply:
yes we don't need this